Fix LayoutTest transitions/flex-transitions.html by syncing from Blink / Chromium upstream

Ahmad-S792 opened the pull request:
#### 744d1201a37974e2c5235684ddf3a83659dbe6ae
<pre>
Fix LayoutTest transitions/flex-transitions.html by syncing from Blink / Chromium upstream

<a href="https://bugs.webkit.org/show_bug.cgi?id=125894">https://bugs.webkit.org/show_bug.cgi?id=125894</a>

Reviewed by Antoine Quint.

Merge: <a href="https://chromium.googlesource.com/chromium/blink/+/40cc24abe770f2fd20d9a09c60cf87502c28063a">https://chromium.googlesource.com/chromium/blink/+/40cc24abe770f2fd20d9a09c60cf87502c28063a</a> &amp;
<a href="https://source.chromium.org/chromium/chromium/src/+/cb4809f7770d86811559c3f39ed0cbc308a73887">https://source.chromium.org/chromium/chromium/src/+/cb4809f7770d86811559c3f39ed0cbc308a73887</a>

Update the expected results to account for the fact that flex-grow and
flex-shrink don&apos;t interpolate smoothly to or from zero. Also update the test to
use values of similar magnitudes for each part of the property shorthand, to
allow the tolerance to be applied meaningfully.

* LayoutTests/transitions/flex-transitions.html: Updated
* LayoutTests/transitions/flex-transitions-expected.txt: Ditto
* LayoutTests/TestExpectations: Remove the test expectation

Canonical link: <a href="https://commits.webkit.org/273266@main">https://commits.webkit.org/273266@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/10292b8993f581567febb73ab828a259db4f467a

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/34901 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/13754 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/36943 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/37614 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/31503 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/16154 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/10834 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/30437 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/35433 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/11654 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/31088 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/10190 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/10259 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/31176 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/38868 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/31717 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/31521 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/36285 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/10362 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/8273 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/34263 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/12179 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/30819 "Passed tests") | | 
| [  ~~🛠 🧪 merge~~](https://ews-build.webkit.org/#/builders/19/builds/8006 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/10905 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/11242 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on LayoutTests/transitions/flex-transitions.html:
```diff
@@ -20,31 +21,31 @@
 .final #negative1 {
-    flex: 1 100px;
+    -webkit-flex: 100 100 100px;
 }
```
Is there a reason to use the prefixed version here?

Ahmad-S792 replied:
I personally had no specific reason but just copied Blink's commit. I think I can rename this test to have 'prefixes' and then another test without? It would give coverage to both. Any input? @nt1m

nt1m replied:
The prefixed versions are just aliases of the unprefixed ones, I'd just remove them. There's not much point testing aliases in general.

nt1m commented on LayoutTests/transitions/flex-transitions.html:
```diff
@@ -56,18 +57,22 @@
+      [0.8, 'row1',      '-webkit-flex', [260, 0,     0], 10],
+      [0.8, 'column1',   '-webkit-flex', [20,  0,   160], 10],
+      [0.8, 'negative1', '-webkit-flex', [100, 100,  90], 10],
+      [0.8, 'no-flex1',  '-webkit-flex', [20,  0,     0], 10],
```
Same question?

nt1m commented on LayoutTests/transitions/flex-transitions.html:
```diff
@@ -14,60 +15,69 @@
 
 #flex-row div {
-    flex: 1 0 0px;
+  flex: 100 0 0px;
```
Can you make indentation consistent?

Ahmad-S792 replied:
Ops! Messed-up. Just did all with 4 spaces for CSS style. :-)

Ahmad-S792 replied:
Done in '[10292b8](https://github.com/WebKit/WebKit/commit/10292b8993f581567febb73ab828a259db4f467a)'.

nt1m reviewed (commented):
Looks good thanks

Ahmad-S792 committed:
Fix LayoutTest transitions/flex-transitions.html by syncing from Blink / Chromium upstream

https://bugs.webkit.org/show_bug.cgi?id=125894

Reviewed by Antoine Quint.

Merge: https://chromium.googlesource.com/chromium/blink/+/40cc24abe770f2fd20d9a09c60cf87502c28063a &
https://source.chromium.org/chromium/chromium/src/+/cb4809f7770d86811559c3f39ed0cbc308a73887

Update the expected results to account for the fact that flex-grow and
flex-shrink don't interpolate smoothly to or from zero. Also update the test to
use values of similar magnitudes for each part of the property shorthand, to
allow the tolerance to be applied meaningfully.

* LayoutTests/transitions/flex-transitions.html: Updated
* LayoutTests/transitions/flex-transitions-expected.txt: Ditto
* LayoutTests/TestExpectations: Remove the test expectation

Canonical link: https://commits.webkit.org/273266@main
